source/common/upstream/health_checker_impl.cc

```diff
@@ -61,10 +61,11 @@ HealthCheckerImplBase::HealthCheckerImplBase(const Cluster& cluster,
       stats_(generateStats(cluster.info()->statsScope())), runtime_(runtime), random_(random),
       interval_(PROTOBUF_GET_MS_REQUIRED(config, interval)),
       interval_jitter_(PROTOBUF_GET_MS_OR_DEFAULT(config, interval_jitter, 0)) {
-  cluster_.addMemberUpdateCb([this](const std::vector<HostSharedPtr>& hosts_added,
-                                    const std::vector<HostSharedPtr>& hosts_removed) -> void {
-    onClusterMemberUpdate(hosts_added, hosts_removed);
-  });
+  cluster_.primaryHosts().addMemberUpdateCb(
```

Comment:
In general, I suspect there is going to be duplication now every time we add a member update callback. Could this somehow be added at the cluster interface level to avoid having the duplication everywhere? I'm not exactly sure how this would work and maybe there is no way to avoid this, but worth thinking about?

Reply:
I started with that convenience function and remove it when there were only two users of it.  Most callers of a memberUpdateCallback are doing it on a HostSet not a cluster.  Current users are
OutlierDetecter (which will likely combine), HealthChecker (which will likely combine) ClusterManagerImpl (which requires slightly different callbacks to determine if the update is for primary or failover) and ClusterBaseImpl (ditto).  Given the few users it was fewer lines of code to kill off the helper but if given the data you'd still prefer it I'll add it back in :-)